index.js

```diff
@@ -89,10 +105,23 @@ module.exports = function serialize(obj, options) {
         return str;
     }
 
+    var placeHolderRegExp;
+    if (options.uid) {
+        placeHolderRegExp = buildPlaceHolderRegExp(uid);
+    } else {
+        placeHolderRegExp = PLACE_HOLDER_REGEXP;
+    }
+
     // Replaces all occurrences of function and regexp placeholders in the JSON
     // string with their string representations. If the original value can not
     // be found, then `undefined` is used.
-    return str.replace(PLACE_HOLDER_REGEXP, function (match, type, valueIndex) {
+    return str.replace(placeHolderRegExp, function (match, backSlash, type, valueIndex) {
+        // The placeholder may not be preceded by a backslash. This is to prevent
+        // replacing things like `"a\"@__R-<UID>-0__@"` and thus outputting
+        // invalid JS.
+        if (backSlash) {
```

Comment:
Now with a better source of entropy, is the backslash escape still required, or did you want to leave it for extra protection?